cmd/zpool/zpool_main.c

```diff
@@ -676,6 +676,20 @@ zpool_do_add(int argc, char **argv)
 		return (1);
 	}
 
+	/* unless manually specified use "ashift" pool property for new vdevs */
+	if (!nvlist_exists(props, ZPOOL_CONFIG_ASHIFT)) {
```

Comment:
Since someone in IRC asked, I will elaborate that I am referring to `zprop_register_number()` when I mention how this setting is a one time thing:

https://github.com/zfsonlinux/zfs/blob/master/module/zcommon/zpool_prop.c#L103